File: app/src/main/java/com/example/autoflow/util/NotificationHelper.java
// Updated NotificationHelper.java - Fixed version
package com.example.autoflow.util;

import android.Manifest;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Build;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresPermission;
import androidx.core.app.NotificationCompat;
import androidx.core.app.NotificationManagerCompat;
import com.example.autoflow.MainActivity;

public class NotificationHelper {

    // Notification channel constants
    private static final String CHANNEL_ID_DEFAULT = "autoflow_default";
    private static final String CHANNEL_ID_SCRIPT = "autoflow_script";
    private static final String CHANNEL_ID_ERROR = "autoflow_error";
    private static final String CHANNEL_NAME_DEFAULT = "AutoFlow Notifications";
    private static final String CHANNEL_NAME_SCRIPT = "Script Notifications";
    private static final String CHANNEL_NAME_ERROR = "Error Notifications";

    // Notification ID counters
    private static int notificationIdCounter = 1000;

    /**
     * Initialize notification channels for different types of notifications
     */
    public static void createNotificationChannels(@NonNull Context context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationManager notificationManager =
                    (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);

            if (notificationManager == null) return;

            // Default channel for general notifications
            NotificationChannel defaultChannel = new NotificationChannel(
                    CHANNEL_ID_DEFAULT,
                    CHANNEL_NAME_DEFAULT,
                    NotificationManager.IMPORTANCE_DEFAULT
            );
            defaultChannel.setDescription("General AutoFlow notifications");
            defaultChannel.enableLights(true);
            defaultChannel.setLightColor(Color.BLUE);
            defaultChannel.enableVibration(true);
            notificationManager.createNotificationChannel(defaultChannel);

            // Script execution channel
            NotificationChannel scriptChannel = new NotificationChannel(
                    CHANNEL_ID_SCRIPT,
                    CHANNEL_NAME_SCRIPT,
                    NotificationManager.IMPORTANCE_DEFAULT
            );
            scriptChannel.setDescription("Notifications from script execution");
            scriptChannel.enableLights(true);
            scriptChannel.setLightColor(Color.GREEN);
            scriptChannel.enableVibration(false);
            notificationManager.createNotificationChannel(scriptChannel);

            // Error channel for high-priority error notifications
            NotificationChannel errorChannel = new NotificationChannel(
                    CHANNEL_ID_ERROR,
                    CHANNEL_NAME_ERROR,
                    NotificationManager.IMPORTANCE_HIGH
            );
            errorChannel.setDescription("Error notifications from AutoFlow");
            errorChannel.enableLights(true);
            errorChannel.setLightColor(Color.RED);
            errorChannel.enableVibration(true);
            errorChannel.setVibrationPattern(new long[]{100, 200, 300, 400, 500});
            notificationManager.createNotificationChannel(errorChannel);
        }
    }

    /**
     * Send a basic notification with title and message
     */
    public static void sendNotification(@NonNull Context context,
                                        @NonNull String title,
                                        @NonNull String message) {
        sendNotification(context, title, message, CHANNEL_ID_DEFAULT, NotificationCompat.PRIORITY_DEFAULT);
    }

    /**
     * Send a notification with custom priority
     */
    public static void sendNotification(@NonNull Context context,
                                        @NonNull String title,
                                        @NonNull String message,
                                        int priority) {
        String channelId = (priority == NotificationCompat.PRIORITY_HIGH) ? CHANNEL_ID_ERROR : CHANNEL_ID_DEFAULT;
        sendNotification(context, title, message, channelId, priority);
    }

    /**
     * Send a script-related notification
     */
    public static void sendScriptNotification(@NonNull Context context,
                                              @NonNull String title,
                                              @NonNull String message) {
        sendNotification(context, title, message, CHANNEL_ID_SCRIPT, NotificationCompat.PRIORITY_DEFAULT);
    }

    /**
     * Send an error notification
     */
    public static void sendErrorNotification(@NonNull Context context,
                                             @NonNull String title,
                                             @NonNull String message) {
        sendNotification(context, title, message, CHANNEL_ID_ERROR, NotificationCompat.PRIORITY_HIGH);
    }

    /**
     * Core notification sending method
     */
    private static void sendNotification(@NonNull Context context,
                                         @NonNull String title,
                                         @NonNull String message,
                                         @NonNull String channelId,
                                         int priority) {
        try {
            // Create pending intent to open the app when notification is tapped
            Intent intent = new Intent(context, MainActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);

            PendingIntent pendingIntent = PendingIntent.getActivity(
                    context,
                    0,
                    intent,
                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
                            ? PendingIntent.FLAG_IMMUTABLE
                            : PendingIntent.FLAG_UPDATE_CURRENT
            );

            // Build the notification
            NotificationCompat.Builder builder = new NotificationCompat.Builder(context, channelId)
                    .setSmallIcon(getNotificationIcon(priority))
                    .setContentTitle(title)
                    .setContentText(message)
                    .setPriority(priority)
                    .setContentIntent(pendingIntent)
                    .setAutoCancel(true) // Remove notification when tapped
                    .setStyle(new NotificationCompat.BigTextStyle().bigText(message)); // Expandable text

            // Add sound for high priority notifications
            if (priority == NotificationCompat.PRIORITY_HIGH) {
                builder.setDefaults(Notification.DEFAULT_ALL);
            }

            // Send the notification
            NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
            int notificationId = getNextNotificationId();
            notificationManager.notify(notificationId, builder.build());

        } catch (SecurityException e) {
            // Handle case where notification permissions are denied
            android.util.Log.e("NotificationHelper", "Notification permission denied: " + e.getMessage());
        } catch (Exception e) {
            android.util.Log.e("NotificationHelper", "Error sending notification: " + e.getMessage());
        }
    }

    /**
     * Send a rich notification with custom actions - FIXED VERSION
     */
    @RequiresPermission(Manifest.permission.POST_NOTIFICATIONS)
    public static void sendRichNotification(@NonNull Context context,
                                            @NonNull String title,
                                            @NonNull String message,
                                            @NonNull String actionText,
                                            @NonNull PendingIntent actionIntent) {
        try {
            // Default app open intent
            Intent intent = new Intent(context, MainActivity.class);
            PendingIntent pendingIntent = PendingIntent.getActivity(
                    context,
                    0,
                    intent,
                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
                            ? PendingIntent.FLAG_IMMUTABLE
                            : PendingIntent.FLAG_UPDATE_CURRENT
            );

            // Build notification with custom action - USING SYSTEM ICONS
            NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID_DEFAULT)
                    .setSmallIcon(getNotificationIcon(NotificationCompat.PRIORITY_DEFAULT))
                    .setContentTitle(title)
                    .setContentText(message)
                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                    .setContentIntent(pendingIntent)
                    .setAutoCancel(true)
                    // FIXED: Using system drawable instead of custom ic_action
                    .addAction(android.R.drawable.ic_media_play, actionText, actionIntent)
                    .setStyle(new NotificationCompat.BigTextStyle().bigText(message));

            NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
            notificationManager.notify(getNextNotificationId(), builder.build());

        } catch (Exception e) {
            Log.e("NotificationHelper", "Error sending rich notification: " + e.getMessage());
        }
    }

    /**
     * Send notification with multiple action buttons
     */
    @RequiresPermission(Manifest.permission.POST_NOTIFICATIONS)
    public static void sendActionNotification(@NonNull Context context,
                                              @NonNull String title,
                                              @NonNull String message,
                                              @NonNull String action1Text,
                                              @NonNull PendingIntent action1Intent,
                                              @NonNull String action2Text,
                                              @NonNull PendingIntent action2Intent) {
        try {
            Intent intent = new Intent(context, MainActivity.class);
            PendingIntent pendingIntent = PendingIntent.getActivity(
                    context,
                    0,
                    intent,
                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
                            ? PendingIntent.FLAG_IMMUTABLE
                            : PendingIntent.FLAG_UPDATE_CURRENT
            );

            NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID_DEFAULT)
                    .setSmallIcon(getNotificationIcon(NotificationCompat.PRIORITY_DEFAULT))
                    .setContentTitle(title)
                    .setContentText(message)
                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                    .setContentIntent(pendingIntent)
                    .setAutoCancel(true)
                    // Using system drawables for action buttons
                    .addAction(android.R.drawable.ic_menu_call, action1Text, action1Intent)
                    .addAction(android.R.drawable.ic_menu_close_clear_cancel, action2Text, action2Intent)
                    .setStyle(new NotificationCompat.BigTextStyle().bigText(message));

            NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
            notificationManager.notify(getNextNotificationId(), builder.build());

        } catch (Exception e) {
            Log.e("NotificationHelper", "Error sending action notification: " + e.getMessage());
        }
    }

    /**
     * Cancel a specific notification
     */
    public static void cancelNotification(@NonNull Context context, int notificationId) {
        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
        notificationManager.cancel(notificationId);
    }

    /**
     * Cancel all notifications from this app
     */
    public static void cancelAllNotifications(@NonNull Context context) {
        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
        notificationManager.cancelAll();
    }

    /**
     * Get appropriate notification icon based on priority - USING SYSTEM ICONS
     */
    private static int getNotificationIcon(int priority) {
        switch (priority) {
            case NotificationCompat.PRIORITY_HIGH:
            case NotificationCompat.PRIORITY_MAX:
                return android.R.drawable.stat_notify_error; // Error/warning icon
            case NotificationCompat.PRIORITY_LOW:
            case NotificationCompat.PRIORITY_MIN:
                return android.R.drawable.ic_dialog_info; // Info icon
            default:
                return android.R.drawable.ic_dialog_email; // Default notification icon
        }
    }

    /**
     * Generate unique notification IDs
     */
    private static synchronized int getNextNotificationId() {
        return notificationIdCounter++;
    }

    /**
     * Check if notifications are enabled for the app
     */
    public static boolean areNotificationsEnabled(@NonNull Context context) {
        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
        return notificationManager.areNotificationsEnabled();
    }

    /**
     * Helper method to convert priority string to integer
     */
    public static int getPriorityFromString(@NonNull String priorityString) {
        switch (priorityString.toLowerCase()) {
            case "low":
                return NotificationCompat.PRIORITY_LOW;
            case "high":
                return NotificationCompat.PRIORITY_HIGH;
            case "max":
                return NotificationCompat.PRIORITY_MAX;
            case "min":
                return NotificationCompat.PRIORITY_MIN;
            default:
                return NotificationCompat.PRIORITY_DEFAULT;
        }
    }

    /**
     * Get system icon resource ID by name for flexibility
     */
    public static int getSystemIcon(@NonNull String iconName) {
        switch (iconName.toLowerCase()) {
            case "play":
                return android.R.drawable.ic_media_play;
            case "pause":
                return android.R.drawable.ic_media_pause;
            case "stop":
                return android.R.drawable.ic_menu_close_clear_cancel;
            case "call":
                return android.R.drawable.ic_menu_call;
            case "settings":
                return android.R.drawable.ic_menu_preferences;
            case "info":
                return android.R.drawable.ic_dialog_info;
            case "alert":
                return android.R.drawable.ic_dialog_alert;
            case "email":
                return android.R.drawable.ic_dialog_email;
            default:
                return android.R.drawable.ic_dialog_info;
        }
    }
}

File: app/src/main/java/com/example/autoflow/util/NotificationHelper.kt
package com.example.autoflow.util

import android.Manifest
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import android.util.Log
import androidx.core.app.ActivityCompat
import androidx.core.app.NotificationCompat
import androidx.core.app.NotificationManagerCompat

/**
 * ✅ ENHANCED NotificationHelper - Complete Version
 * Combines existing functionality with Meeting Mode features
 * Handles notification channels, creation, and display with backward compatibility
 */
object NotificationHelper {

    private const val TAG = "NotificationHelper"
    private const val DEFAULT_NOTIFICATION_ID = 1000

    // ✅ EXISTING + NEW notification channels (merged)
    private const val CHANNEL_ID_DEFAULT = "autoflow_default"
    private const val CHANNEL_ID_HIGH_PRIORITY = "autoflow_high_priority"
    private const val CHANNEL_ID_LOW_PRIORITY = "autoflow_low_priority"
    private const val CHANNEL_ID_SCRIPT = "autoflow_script"
    private const val CHANNEL_ID_ERROR = "autoflow_error"

    // ✅ NEW Meeting Mode channels
    private const val CHANNEL_GENERAL = "autoflow_general"
    private const val CHANNEL_AUTO_REPLY = "autoflow_auto_reply"
    private const val CHANNEL_MEETING_MODE = "autoflow_meeting_mode"

    // Channel names
    private const val CHANNEL_NAME_DEFAULT = "AutoFlow Notifications"
    private const val CHANNEL_NAME_HIGH = "Important AutoFlow Alerts"
    private const val CHANNEL_NAME_LOW = "AutoFlow Updates"
    private const val CHANNEL_NAME_SCRIPT = "Script Notifications"
    private const val CHANNEL_NAME_ERROR = "Error Notifications"

    // ✅ COMPATIBLE Priority constants
    const val PRIORITY_LOW = NotificationCompat.PRIORITY_LOW
    const val PRIORITY_HIGH = NotificationCompat.PRIORITY_HIGH
    const val PRIORITY_DEFAULT = NotificationCompat.PRIORITY_DEFAULT

    /**
     * ✅ ENHANCED: Create all notification channels (both existing + new Meeting Mode)
     */
    fun createNotificationChannels(context: Context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager ?: return

            val channels = listOf(
                // ✅ EXISTING channels (enhanced)
                NotificationChannel(
                    CHANNEL_ID_DEFAULT,
                    CHANNEL_NAME_DEFAULT,
                    NotificationManager.IMPORTANCE_DEFAULT
                ).apply {
                    description = "General AutoFlow notifications"
                    enableVibration(true)
                    enableLights(true)
                },

                NotificationChannel(
                    CHANNEL_ID_HIGH_PRIORITY,
                    CHANNEL_NAME_HIGH,
                    NotificationManager.IMPORTANCE_HIGH
                ).apply {
                    description = "Important AutoFlow alerts"
                    enableVibration(true)
                    enableLights(true)
                },

                NotificationChannel(
                    CHANNEL_ID_LOW_PRIORITY,
                    CHANNEL_NAME_LOW,
                    NotificationManager.IMPORTANCE_LOW
                ).apply {
                    description = "AutoFlow status updates"
                    enableVibration(false)
                    enableLights(false)
                },

                NotificationChannel(
                    CHANNEL_ID_SCRIPT,
                    CHANNEL_NAME_SCRIPT,
                    NotificationManager.IMPORTANCE_DEFAULT
                ).apply {
                    description = "Notifications from user scripts"
                    enableVibration(true)
                },

                NotificationChannel(
                    CHANNEL_ID_ERROR,
                    CHANNEL_NAME_ERROR,
                    NotificationManager.IMPORTANCE_HIGH
                ).apply {
                    description = "Error notifications"
                    enableVibration(true)
                    enableLights(true)
                },

                // ✅ NEW Meeting Mode channels
                NotificationChannel(
                    CHANNEL_GENERAL,
                    "General Notifications",
                    NotificationManager.IMPORTANCE_DEFAULT
                ).apply {
                    description = "General AutoFlow notifications (Meeting Mode compatible)"
                },

                NotificationChannel(
                    CHANNEL_AUTO_REPLY,
                    "Auto-Reply Notifications",
                    NotificationManager.IMPORTANCE_LOW
                ).apply {
                    description = "Notifications when auto-reply SMS is sent"
                },

                NotificationChannel(
                    CHANNEL_MEETING_MODE,
                    "Meeting Mode",
                    NotificationManager.IMPORTANCE_HIGH
                ).apply {
                    description = "Meeting mode status notifications"
                }
            )

            notificationManager.createNotificationChannels(channels)
            Log.d(TAG, "✅ All notification channels created (${channels.size} channels)")
        }
    }

    /**
     * ✅ PRIMARY: Enhanced sendNotification method with full compatibility
     * Supports both existing and Meeting Mode calls
     */
    @JvmStatic
    @JvmOverloads
    fun sendNotification(
        context: Context,
        title: String,
        message: String,
        priority: Int = NotificationCompat.PRIORITY_DEFAULT,
        notificationId: Int = DEFAULT_NOTIFICATION_ID,
        channelId: String? = null // ✅ Optional for backward compatibility
    ) {
        if (!hasNotificationPermission(context)) {
            Log.w(TAG, "⚠️ Notification permission not granted")
            return
        }

        try {
            // ✅ Smart channel selection
            val targetChannelId = channelId ?: getChannelIdForIntPriority(priority)

            val builder = NotificationCompat.Builder(context, targetChannelId)
                .setSmallIcon(getNotificationIcon())
                .setContentTitle(title)
                .setContentText(message)
                .setPriority(priority)
                .setAutoCancel(true)
                .setStyle(NotificationCompat.BigTextStyle().bigText(message))

            // ✅ Add content intent to open app
            val intent = context.packageManager.getLaunchIntentForPackage(context.packageName)
            if (intent != null) {
                val pendingIntent = PendingIntent.getActivity(
                    context,
                    0,
                    intent,
                    PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
                )
                builder.setContentIntent(pendingIntent)
            }

            try {
                NotificationManagerCompat.from(context).notify(notificationId, builder.build())
                Log.d(TAG, "✅ Notification sent: $title")
            } catch (e: SecurityException) {
                Log.e(TAG, "❌ SecurityException: Missing notification permission", e)
            }

        } catch (e: Exception) {
            Log.e(TAG, "❌ Error sending notification", e)
        }
    }

    /**
     * ✅ EXISTING: Enhanced notification with custom action
     */
    @JvmStatic
    fun sendNotificationWithAction(
        context: Context,
        title: String,
        message: String,
        actionTitle: String,
        actionIntent: PendingIntent,
        priority: Int = NotificationCompat.PRIORITY_DEFAULT,
        notificationId: Int = DEFAULT_NOTIFICATION_ID
    ) {
        if (!hasNotificationPermission(context)) {
            Log.w(TAG, "⚠️ Notification permission not granted")
            return
        }

        try {
            val channelId = getChannelIdForIntPriority(priority)

            val builder = NotificationCompat.Builder(context, channelId)
                .setSmallIcon(getNotificationIcon())
                .setContentTitle(title)
                .setContentText(message)
                .setPriority(priority)
                .setAutoCancel(true)
                .setStyle(NotificationCompat.BigTextStyle().bigText(message))
                .addAction(0, actionTitle, actionIntent)

            try {
                NotificationManagerCompat.from(context).notify(notificationId, builder.build())
                Log.d(TAG, "✅ Notification with action sent: $title")
            } catch (e: SecurityException) {
                Log.e(TAG, "❌ SecurityException: Missing notification permission", e)
            }

        } catch (e: Exception) {
            Log.e(TAG, "❌ Error sending notification", e)
        }
    }

    /**
     * ✅ EXISTING: Script notification (for user scripts)
     */
    @JvmStatic
    fun sendScriptNotification(
        context: Context,
        title: String,
        message: String,
        notificationId: Int = (System.currentTimeMillis() % 10000).toInt()
    ) {
        if (!hasNotificationPermission(context)) {
            Log.w(TAG, "⚠️ Notification permission not granted")
            return
        }

        try {
            val builder = NotificationCompat.Builder(context, CHANNEL_ID_SCRIPT)
                .setSmallIcon(getNotificationIcon())
                .setContentTitle(title)
                .setContentText(message)
                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                .setAutoCancel(true)
                .setStyle(NotificationCompat.BigTextStyle().bigText(message))

            try {
                NotificationManagerCompat.from(context).notify(notificationId, builder.build())
                Log.d(TAG, "✅ Script notification sent: $title")
            } catch (e: SecurityException) {
                Log.e(TAG, "❌ SecurityException: Missing notification permission", e)
            }

        } catch (e: Exception) {
            Log.e(TAG, "❌ Error sending script notification", e)
        }
    }

    /**
     * ✅ EXISTING: Send error notification
     */
    @JvmStatic
    fun sendErrorNotification(
        context: Context,
        title: String,
        message: String,
        notificationId: Int = (System.currentTimeMillis() % 10000).toInt()
    ) {
        if (!hasNotificationPermission(context)) {
            Log.w(TAG, "⚠️ Notification permission not granted")
            return
        }

        try {
            val builder = NotificationCompat.Builder(context, CHANNEL_ID_ERROR)
                .setSmallIcon(android.R.drawable.stat_notify_error)
                .setContentTitle(title)
                .setContentText(message)
                .setPriority(NotificationCompat.PRIORITY_HIGH)
                .setAutoCancel(true)
                .setStyle(NotificationCompat.BigTextStyle().bigText(message))

            try {
                NotificationManagerCompat.from(context).notify(notificationId, builder.build())
                Log.d(TAG, "✅ Error notification sent: $title")
            } catch (e: SecurityException) {
                Log.e(TAG, "❌ SecurityException: Missing notification permission", e)
            }

        } catch (e: Exception) {
            Log.e(TAG, "❌ Error sending error notification", e)
        }
    }

    /**
     * ✅ EXISTING: Send progress notification
     */
    @JvmStatic
    fun sendProgressNotification(
        context: Context,
        title: String,
        message: String,
        progress: Int,
        maxProgress: Int = 100,
        notificationId: Int = DEFAULT_NOTIFICATION_ID
    ) {
        if (!hasNotificationPermission(context)) return

        try {
            val builder = NotificationCompat.Builder(context, CHANNEL_ID_DEFAULT)
                .setSmallIcon(getNotificationIcon())
                .setContentTitle(title)
                .setContentText(message)
                .setPriority(NotificationCompat.PRIORITY_LOW)
                .setProgress(maxProgress, progress, false)
                .setOngoing(true)

            try {
                NotificationManagerCompat.from(context).notify(notificationId, builder.build())
            } catch (e: SecurityException) {
                Log.e(TAG, "❌ SecurityException: Missing notification permission", e)
            }

        } catch (e: Exception) {
            Log.e(TAG, "❌ Error sending progress notification", e)
        }
    }

    /**
     * ✅ NEW: Meeting Mode specific notifications
     */
    @JvmStatic
    fun sendMeetingModeNotification(
        context: Context,
        title: String,
        message: String,
        notificationId: Int = (System.currentTimeMillis() % 10000).toInt()
    ) {
        sendNotification(
            context = context,
            title = title,
            message = message,
            priority = NotificationCompat.PRIORITY_HIGH,
            notificationId = notificationId,
            channelId = CHANNEL_MEETING_MODE
        )
    }

    /**
     * ✅ NEW: Auto-reply specific notifications
     */
    @JvmStatic
    fun sendAutoReplyNotification(
        context: Context,
        phoneNumber: String,
        message: String,
        notificationId: Int = (System.currentTimeMillis() % 10000).toInt()
    ) {
        sendNotification(
            context = context,
            title = "📱 Auto-Reply Sent",
            message = "Replied to $phoneNumber: $message",
            priority = NotificationCompat.PRIORITY_LOW,
            notificationId = notificationId,
            channelId = CHANNEL_AUTO_REPLY
        )
    }

    /**
     * ✅ EXISTING: Cancel a notification
     */
    @JvmStatic
    fun cancelNotification(context: Context, notificationId: Int) {
        try {
            NotificationManagerCompat.from(context).cancel(notificationId)
            Log.d(TAG, "✅ Notification cancelled: $notificationId")
        } catch (e: Exception) {
            Log.e(TAG, "❌ Error cancelling notification", e)
        }
    }

    /**
     * ✅ EXISTING: Cancel all notifications
     */
    @JvmStatic
    fun cancelAllNotifications(context: Context) {
        try {
            NotificationManagerCompat.from(context).cancelAll()
            Log.d(TAG, "✅ All notifications cancelled")
        } catch (e: Exception) {
            Log.e(TAG, "❌ Error cancelling all notifications", e)
        }
    }

    /**
     * ✅ EXISTING: Check if notifications are enabled
     */
    @JvmStatic
    fun areNotificationsEnabled(context: Context): Boolean {
        return NotificationManagerCompat.from(context).areNotificationsEnabled()
    }

    // ✅ PRIVATE HELPER METHODS

    /**
     * ✅ NEW: Smart channel selection for Int priority
     */
    private fun getChannelIdForIntPriority(priority: Int): String {
        return when (priority) {
            NotificationCompat.PRIORITY_HIGH, NotificationCompat.PRIORITY_MAX -> CHANNEL_ID_HIGH_PRIORITY
            NotificationCompat.PRIORITY_LOW, NotificationCompat.PRIORITY_MIN -> CHANNEL_ID_LOW_PRIORITY
            else -> CHANNEL_ID_DEFAULT
        }
    }

    /**
     * ✅ EXISTING: String priority support (backward compatibility)
     */
    private fun getChannelId(priority: String): String {
        return when (priority.uppercase()) {
            "HIGH" -> CHANNEL_ID_HIGH_PRIORITY
            "LOW" -> CHANNEL_ID_LOW_PRIORITY
            else -> CHANNEL_ID_DEFAULT
        }
    }

    /**
     * ✅ EXISTING: String to Int priority conversion
     */
    private fun getPriorityInt(priority: String): Int {
        return when (priority.uppercase()) {
            "HIGH" -> NotificationCompat.PRIORITY_HIGH
            "LOW" -> NotificationCompat.PRIORITY_LOW
            else -> NotificationCompat.PRIORITY_DEFAULT
        }
    }

    /**
     * ✅ EXISTING: Get notification icon
     */
    private fun getNotificationIcon(): Int {
        // You can replace with your app's actual icon resource
        return android.R.drawable.ic_dialog_info // Placeholder
    }

    /**
     * ✅ EXISTING: Check notification permission
     */
    private fun hasNotificationPermission(context: Context): Boolean {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            ActivityCompat.checkSelfPermission(
                context,
                Manifest.permission.POST_NOTIFICATIONS
            ) == PackageManager.PERMISSION_GRANTED
        } else {
            true
        }
    }
}
